posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc

```diff
@@ -0,0 +1,413 @@
+
+== MicroProfile Applications
+Many developers use MicroProfile specifications for configuring, securing, and observing cloud native applications. MicroProfile offers a set of standards APIs for cloud-native applications that free your them from vendor lock-in. Open Liberty is the leading implementation for MicroProfile specifications.
+
+As we enter the serverless era, it is very important to get MicroProfile applications running in a serverless environment. Luckily, it is very straightforward to get these applications running as functions in an environment that supports CloudEvents, such as IBM Cloud Code Engine. In the following sections, we demonstrate how to get a MicroProfile application that is deployed to Open Liberty running as a function in IBM Cloud Code Engine with CloudEvent.
+
+== Run MicroProfile applications  as a serverless function in Code Engine
+RESTful MicroProfile Applications can run seamlessly as a serverless function in Code Engine with CloudEvent. This section explains how.
+
+=== Prerequisites
+Before you start you will need the following prerequisites:
+
+* An IBM Cloud Account
+* Docker
+* An externally accessible container Registry that can host your images (IBM Cloud Container Registry is available as part of your cloud account)
+
+=== Creating your MicroProfile application
+If you don't have a suitable RESTful application to host as a set of functions you can use the https://openliberty.io/guides/getting-started.html[OpenLiberty Getting Started guide] to get a simple MicroProfile application that is  well suited to a serverless environment, such as IBM Cloud Code Engine.
+
+If using the Getting Started code and you already have experience with OpenLiberty and MicroProfile, you can use the `finish` folder as your starting point and go to the next step. If you do not, complete the guide up to and including the https://openliberty.io/guides/getting-started.html#running-the-application-in-a-docker-container[Running the application in a Docker container] step to generate the application you'll use in this post.
+
+
+After you have the application, the next step is to  test it to ensure that it is running and accessible.
+
+Run the following command to run your container locally on its `http` port:
+```
+docker run -p 9080:9080 openliberty-getting-started:1.0-SNAPSHOT
+```
+
+To invoke the application, you can use either your browser, curl, or an API testing tool. For example you can use the following curl command:
+```
+curl http://localhost:9080/dev/system/properties
+```
+This command invokes the `getProperties()` method in the application's `SystemResource` class
+
+=== Deploying your application to a container registry to prepare for IBM Cloud Code Engine
+Having sucessfully built and tested your image locally, you now need to put it somewhere that IBM Cloud Code Engine can get to it.
+
+ICCE supports two primary methods to get your code into a runnable format
```

Comment:
here also, you might simplify by focusing on just ICCR, mention the other options but dont discuss

Reply:
updated